### DISCUSSION TITLE
For flutter development, how to reselect the project root when I accidently selected the wrong one on the first time?

### DISCUSSION THREAD

**Discussion by sylecn** [emacs-lsp/lsp-mode#3987] [Q&A] [answered]
I installed lsp-dart, I visited lib/main.dart, lsp was started and it asked me the project root. I accepted the default one, which was the lib/ directory, which I later know is wrong.
Now whenever I visit that file, lsp will either auto start or reuse existing analysis_server for that directory. I can't find where this information is cached.

How would I re-select the project root?

**Comment by yyoncho** [comment, marked as answer]
Use M-x lsp-workspace-folders-add/remove

**Reply by yyoncho** [reply]
FWIW a lot of functionality that you need/use is available when you press right click:
![image](https://user-images.githubusercontent.com/13259670/223989435-ddf13151-622b-45f9-aaf7-7856ba19be9d.png)

**Comment by sylecn** [comment]
Oh, that was fast. Thanks.

I even checked the menu-bar, didn't check context-menu.
